Classes/Exporter/EventsExporter.php

```diff
@@ -0,0 +1,390 @@
+            $downloadLinks[] = $this->buildLink($linkForDownload, $storagePages);
+        }
+
+        $eventRecord['download_links'] = $downloadLinks;
+    }
+
+    /**
+     * Since using typolink for URIs we can not use $this->contentObjectRenderer->typoLink_URL anymore, as
+     * it needs a domain name which we do not have in CLI mode. Further we have installed jumpurl which hooks
+     * into typolink process and breaks at invalid HTTP_HOST.
+     * That's why we have to build our own URI.
+     */
+    protected function buildLink(?Link $link, array $storagePages = []): array
+    {
+        if ($link === null) {
+            return [];
+        }
+
+        $linkRecord = [
+            'uid' => $link->getUid(),
+            'crdate' => $this->formatDateToISO($link->getCrdate()),
+            'tstamp' => $this->formatDateToISO($link->getTstamp()),
+            'starttime' => $link->getStarttime(),
+            'endtime' => $link->getEndtime(),
+            'hidden' => $link->getHidden(),
+            'deleted' => $link->getDeleted(),
+            'title' => $link->getTitle(),
+        ];
+
+        try {
+            $linkInformation = $this->linkService->resolve($link->getLink());
+            switch ($linkInformation['type']) {
+                case 'url':
+                    $linkRecord['link'] = $linkInformation['url'];
+                    break;
+                case 'file':
+                    $baseUrl = $this->getBaseUrlForImages($storagePages);
+                    $linkRecord['link'] = $baseUrl . '/' . $linkInformation['file']->getPublicUrl();
+                    break;
+                default:
+                    $linkRecord['link'] = '';
+                    break;
+            }
+        } catch (UnknownLinkHandlerException $e) {
+            $linkRecord = [];
+        }
+
+        return $linkRecord;
+    }
+
+    protected function getFormattedHourMinute(\DateTimeImmutable $eventBegin, string $time): string
+    {
+        $time = trim($time);
+        if ($time === '' || !strpos($time, ':')) {
+            return '';
+        }
+
+        [$hour, $minute] = explode(':', $time);
+
+        return $eventBegin->format('Ymd') . $hour . $minute . '00';
+    }
+
+    protected function getBaseUrlForImages(array $storagePages): string
+    {
+        // Use first storage page to retrieve the public base URL
+        $firstStoragePage = (int)current($storagePages);
+        try {
+            $site = $this->siteFinder->getSiteByPageId($firstStoragePage);
+
+            // getBase may add query (?) and fragment (#). Remove them from URL
+            $baseUrl = (string)$site->getBase();
+            $pos = strpos($baseUrl, '?');
+            if ($pos !== false) {
+                $baseUrl = substr($baseUrl, 0, $pos);
+            }
+
+            return $baseUrl;
+        } catch (SiteNotFoundException $e) {
+        }
+
+        return '';
+    }
+
+    protected function formatDateToISO(?\DateTimeImmutable $date): string
```

Comment:
@sfroemkenjw only suggestion. The functions formatDateToISO, getBaseUrlForImages, createTimeRecord looks more like utility functions. It would be nice if a Utility class handled these.